src/sidepanel/v2/hooks/useMessageHandler.ts

```
@@ -333,4 +383,77 @@ export function useMessageHandler() {
       streamingMessages.current.clear()
     }
   }, [addMessageListener, removeMessageListener, handleStreamUpdate, handleWorkflowStatus])
+  
+  // PDF.js parsing handler in side panel context
+  useEffect(() => {
+    const handler = (request: any, _sender: chrome.runtime.MessageSender, sendResponse: (resp?: any) => void) => {
+      if (request?.type !== MessageType.PDF_PARSE_REQUEST) return
+      ;(async () => {
+        try {
+          const { url, maxPages = 40 } = request.payload || {}
+          let pdfUrl = url
```

ESLint

[error] 'pdfUrl' is never reassigned. Use 'const' instead.

(prefer-const)